### PR TITLE
Add `camlidl.opam`, `META`, and `camlidl.install` files (for 1.12)

### PR DESCRIPTION
**Pull request by MisterDA**
This helps by allowing `opam pin` to work on the repository (opam needs the opam file). It does increase the maintenance a bit because releasing a new version means bumping the version numbers from the opam and META files. The META and `camlidl.install` could be generated by switching or adding Dune build system support.
The files were copied from the current opam-repository.

### CODE REVIEW

File: camlidl.opam [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+build: [
+  ["mv" "config/Makefile.unix" "config/Makefile"] {os != "win32"}
+  ["mv" "config/Makefile.mingw" "config/Makefile"] {os = "win32"}
+  [make "all"]
+]
+dev-repo: "git+https://github.com/xavierleroy/camlidl.git"
+extra-files: [
+  ["camlidl.install" "md5=cf56e14faed046880b7c9d0f4cd737f1"]
+  ["META" "md5=2e2f78ac8c3e9f5581a28b2cec6d755b"]
+]
```
Comment (xavierleroy) [inline camlidl.opam]: Do I understand correctly that at every release, `META` will change (to bump the version number), and therefore those checksums must be updated manually?

Is there a way NOT to store these checksums in the .opam file?  After all, we don't checksum individual source files, so why checksum these two?

**Review by xavierleroy — COMMENTED** [review]
Thanks for this proposal, and apologies for taking so long to look at it.  I'm globally in favor, but there's one aspect I really dislike, see below.

File: camlidl.opam [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+CamlIDL is a stub code generator for OCaml. It automates the
+generation of C stub code required for the Caml/C interface, based on
+an MIDL specification. Also provides some support for Microsoft's COM
+software components."""
+maintainer: "Nicolas Berthier <m@nberth.space>"
+authors: "Xavier Leroy"
+license: [
+  "QPL-1.0 WITH OCaml-LGPL-linking-exception"
+  "LGPL-2.0-or-later WITH OCaml-LGPL-linking-exception"
+]
```
Comment (xavierleroy) [inline camlidl.opam]: For the record: the QPL is no longer used, CamlIDL as a whole is under the LGPL-2-0 with OCaml exception.

File: camlidl.opam (hunk 1 above)
Comment (MisterDA) [reply camlidl.opam]: Looking at opam's manual for [`extra-files`](https://opam.ocaml.org/doc/Manual.html#opamfield-extra-files), it seems that the checksums are required for this field. I'll ask around what can be done instead.

Comment (MisterDA) [reply camlidl.opam]: Ah, I copied the files from opam--that's why it needed the `extra-files`, as they're not distributed with the source tarball; but if we include them in this repo there's no need for this field.

**Comment by xavierleroy** [comment]
OK, thanks for the additional explanations.  I did a manual merge + update of this PR here: https://github.com/xavierleroy/camlidl/commit/1c1e87e3f56c2c6b3226dd0af3510ef414b462d0 .  Let me know if you see anything strange.

**Pull request closed without merging**